src/test/java/com/uid2/admin/vertx/SearchServiceTest.java

```diff
@@ -0,0 +1,597 @@
+package com.uid2.admin.vertx;
+
+import com.uid2.admin.auth.AdminUser;
+import com.uid2.admin.vertx.service.IService;
+import com.uid2.admin.vertx.service.SearchService;
+import com.uid2.admin.vertx.test.ServiceTestBase;
+import com.uid2.shared.auth.ClientKey;
+import com.uid2.shared.auth.OperatorKey;
+import com.uid2.shared.auth.Role;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.junit5.VertxTestContext;
+import org.instancio.Instancio;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.instancio.Select.field;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class SearchServiceTest extends ServiceTestBase {
+    private final static String searchUrl = "api/search?keyOrSecret=";
+
+    @Override
+    protected IService createService() {
+        return new SearchService(auth, clientKeyProvider, operatorKeyProvider, adminUserProvider);
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Role.class, names = {"ADMINISTRATOR"}, mode = EnumSource.Mode.EXCLUDE)
+    void searchAsNonAdminFails(Role role, Vertx vertx, VertxTestContext testContext) {
+        fakeAuth(role);
+        get(vertx, searchUrl, expectHttpError(testContext, 401));
+    }
+
+    @Test
+    void searchAsAdminPasses(Vertx vertx, VertxTestContext testContext) {
+        fakeAuth(Role.ADMINISTRATOR);
+        get(vertx, searchUrl + "123456", response -> {
+            assertTrue(response.succeeded());
```

Comment:
We can use `assertAll()` instead of asserting one by one - this would allow us to catch all errors at once, rather than the errors coming up one by one

This would apply for the rest of the test cases here

Reply:
e.g.

```
assertAll(
  "HTTP response assertion",
  () -> assertTrue(response.succeeded()),
  () -> assertEquals(200, httpResponse.statusCode())
);
```

Reply:
Good idea. Done